### DISCUSSION TITLE
Is it possible to export a om# score into openmusic, in order to be able to export it to other formats?

### DISCUSSION THREAD

**Discussion by smoge** [cac-t-u-s/om-sharp#357] [Q&A] [answered]
Is it possible to export a om# score into openmusic, in order to be able to export it to other formats?

**Comment by j-bresson** [comment]
There is currently no OM# -> OM export possibilities unfortunately 😞 
The only workaround is to export in MIDI and re-quantize in OM.

**Comment by smoge** [comment]
@j-bresson is there some documentation on the format of OM# objects, like voices? 

I would like to try to generate the code in haskell or supercollider. It's totally possible with OM objects, I believe with the correct information, it won't be a problem to implement it with OM#. Just point out to me where to find it.

For OM objects, I could, for example, try this as a Haskell module (using quasiquotations in this case):

```

module OM where

import Data.Text (Text)
import qualified Data.Text.IO as T
import NeatInterpolation (trimming)

header :: Text
header =
  [trimming|
;fileheader
; (7.0 :inst 0 0 0 \"doc\" 183)
;endfileheader

(in-package :om)
|]

poly :: Text -> Text
poly a =
  [trimming|
(setf *instance-to-load*
(omng-make-new-instance
(make-instance 'poly
  :voices
  (list
    $a 
  ))

"instance"))
|]
```

**Reply by j-bresson** [reply, marked as answer]
OM# storage format is supposed to be fairly readable: you can look at it by creating a simple patch with the object, and open the patch file as text (you can use the File / "Open as Text…" menu to do that within OM#). 

In order to get the formatted content of just a specific object, you can connect it to `omng-save`: this is the function called internally to format the contents in text files. You can then route the output of `omng-save` to a [`TEXTBUFFER`](https://cac-t-u-s.github.io/pages/textbuffer) and use `save-as-text` to export! 

<img width="269" alt="voice-export" src="https://github.com/cac-t-u-s/om-sharp/assets/7666142/10ffb3b8-7625-46b9-aae5-95fcc50dc5ff">
